nautobot/utilities/query_functions.py

```diff
@@ -8,8 +9,19 @@ class CollateAsChar(Func):
     Disregard localization by collating a field as a plain character string. Helpful for ensuring predictable ordering.
     """
 
-    function = "C"
-    template = '(%(expressions)s) COLLATE "%(function)s"'
+    function = None
+    template = "(%(expressions)s) COLLATE %(function)s"
+
+    def as_sql(self, compiler, connection, function=None, template=None, arg_joiner=None, **extra_context):
+        engine = connection.settings_dict["ENGINE"]
+        if "postgres" in engine:
+            function = '"C"'
```

Comment:
I'm just curious...  why does this needs to be wrapped in literal double quotes?

Reply:
that's the Postgres syntax for collations: https://www.postgresql.org/docs/9.1/collation.html